app/src/main/java/org/cru/godtools/fragment/ToolDetailsFragment.java

```diff
@@ -280,24 +286,23 @@ private void updateViews() {
             mTitle.setText(ModelUtils.getTranslationName(getContext(), mLatestTranslation, mTool));
         }
         bindShares(mShares, mTool);
-        if (mDescription != null) {
-            mDescription.setText(ModelUtils.getTranslationDescription(getContext(), mLatestTranslation, mTool));
-        }
-        if (mLanguagesHeader != null) {
-            final int count = mLanguages.size();
-            mLanguagesHeader.setText(mLanguagesHeader.getResources()
-                                             .getQuantityString(R.plurals.label_tools_languages, count, count));
-        }
-        if (mLanguagesView != null) {
-            mLanguagesView.setVisibility(mLanguages.isEmpty() ? View.GONE : View.VISIBLE);
-            mLanguagesView.setText(Stream.of(mLanguages)
-                                           .map(l -> LocaleUtils.getDisplayName(l, mLanguagesView.getContext(), null,
-                                                                                null))
-                                           .withoutNulls()
-                                           .sorted(String.CASE_INSENSITIVE_ORDER)
-                                           .reduce((l1, l2) -> l1 + ", " + l2)
-                                           .orElse(""));
+
+        final int count = mLanguages.size();
+        if (mViewPager != null) {
+            ToolDetailsAdapter adapter = new ToolDetailsAdapter();
```

Comment:
`updateViews()` is called whenever backing data changes. creating a brand new Adapter every time data changes is heavy. `updateViews()` should only be manipulating existing views, not recreating a significant part of the view hierarchy.